payments-core/src/main/java/com/stripe/android/StripePaymentController.kt

```diff
@@ -124,12 +114,23 @@ internal class StripePaymentController internal constructor(
         )
     }
 
-    private val stripe3ds2CompletionStarterFactory =
-        { host: AuthActivityStarter.Host, requestCode: Int ->
-            stripe3ds2ChallengeLauncher?.let {
-                Stripe3ds2CompletionStarter.Modern(it)
-            } ?: Stripe3ds2CompletionStarter.Legacy(host, requestCode)
-        }
+    private val authenticatorRegistry: IntentAuthenticatorRegistry =
+        DefaultIntentAuthenticatorRegistry.getInstance(
```

Comment:
So there will be one instance of the DefaultIntentAuthenticator for each StripePaymentController?

Reply:
No, I think this should be a singleton. ATM it's saved as a static instance of `StripePaymentController`, but later when `StripePaymentController` is fully dagerrized, we can inject a singleton `IntentAuthenticator`